idprovider/local/idprovider.go

```diff
@@ -28,57 +28,57 @@ import (
+// IDProvider represents an ID provider that provides access to peer IDs stored locally in a file on the file system.
 //
-// It generates a cache of all contacts in the yaml file during initialization. Read, Write and Delete
-// operations act only on the cached list of contacts and do not update the yaml file.
-// The changes in cache can be updated to the yaml file by explicitly calling UpdateStorage method.
+// It generates a cache of all peer IDs in the ID provider file during initialization. Read, Write and Delete
+// operations act only on the cached list of peer IDs and do not update the ID provider file.
+// The changes in cache can be updated to the ID provider file by explicitly calling UpdateStorage method.
 //
 // It also stores an instance of wallet backend that will be used or decoding address strings.
-type Provider struct {
-	*contactsCache
+type IDProvider struct {
+	*idProviderCache
 
-	contactsFilePath string
+	localFilePath string
 }
 
-// New returns an instance of contacts provider to access the contacts in the given yaml file.
+// NewIDprovider returns an instance of ID provider to access the peer IDs in the given ID provider file.
 //
-// All the contacts are cached in memory during initialization and Read, Write, Delete operations
-// affect only the cache. The changes are updated to the yaml file only when UpdateStorage method
-// is explicitly called. There is no mechanism to reload the cache if the yaml file is updated.
+// All the peer IDs are cached in memory during initialization and Read, Write, Delete operations
+// affect only the cache. The changes are updated to the ID provider file only when UpdateStorage method
+// is explicitly called. There is no mechanism to reload the cache if the ID provider file is updated.
 //
 // Backend is used for decoding the address strings during initialization.
-func New(filePath string, backend perun.WalletBackend) (*Provider, error) {
+func NewIDprovider(filePath string, backend perun.WalletBackend) (*IDProvider, error) {
 	f, err := os.Open(filepath.Clean(filePath))
 	if err != nil {
 		return nil, err
 	}
 	defer f.Close() // nolint: errcheck, gosec  // safe to defer f.Close() for files opened in read mode.
 
-	cache := make(map[string]perun.Peer)
+	cache := make(map[string]perun.PeerID)
 	decoder := yaml.NewDecoder(f)
 	if err = decoder.Decode(&cache); err != nil && err != io.EOF {
 		return nil, err
 	}
 
-	contactsCache, err := newContactsCache(cache, backend)
+	idProviderCache, err := newIDProviderCache(cache, backend)
 	if err != nil {
 		return nil, err
 	}
-	return &Provider{
-		contactsCache:    contactsCache,
-		contactsFilePath: filePath,
+	return &IDProvider{
+		idProviderCache: idProviderCache,
+		localFilePath:   filePath,
 	}, nil
 }
 
-// UpdateStorage writes the latest state of contacts cache to the yaml file.
-func (c *Provider) UpdateStorage() error {
+// UpdateStorage writes the latest state of idprovider cache to the file on the disk.
```

Comment:
```suggestion
// UpdateStorage writes the latest state of ID Provider cache to the file on the disk.
```

Using "ID Provider" as two words in comments or error messages.